pkg/server/commands/check_command.go

```diff
@@ -0,0 +1,161 @@
+package commands
+
+import (
+	"context"
+	"errors"
+	"math"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openfga/openfga/internal/condition"
+	"github.com/openfga/openfga/internal/graph"
+	"github.com/openfga/openfga/internal/validation"
+	"github.com/openfga/openfga/pkg/middleware/validator"
+	serverErrors "github.com/openfga/openfga/pkg/server/errors"
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/storage/storagewrappers"
+	"github.com/openfga/openfga/pkg/tuple"
+	"github.com/openfga/openfga/pkg/typesystem"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+const (
+	defaultResolveNodeLimit           = 25
+	defaultMaxConcurrentReadsForCheck = math.MaxUint32
+)
+
+type CheckQuery struct {
+	logger        logger.Logger
+	checkResolver graph.CheckResolver
+	typesys       *typesystem.TypeSystem
+	datastore     storage.RelationshipTupleReader
+
+	servicename        string
+	methodname         string
+	resolveNodeLimit   uint32
+	maxConcurrentReads uint32
+}
+
+type CheckQueryOption func(*CheckQuery)
+
+func WithCheckCmdServiceAndMethodName(servicename, methodname string) CheckQueryOption {
+	return func(c *CheckQuery) {
+		c.servicename, c.methodname = servicename, methodname
+	}
+}
+
+func WithCheckCmdResolveNodeLimit(nl uint32) CheckQueryOption {
```

Comment:
All other commands _try_ to follow a slightly different convention

```suggestion
func WithCheckQueryResolveNodeLimit(nl uint32) CheckQueryOption {
```